angular/projects/admin-nrpti/src/app/records/administrative-sanctions/administrative-sanction-add-edit/administrative-sanction-add-edit.component.ts

```diff
@@ -457,7 +457,11 @@ export class AdministrativeSanctionAddEditComponent implements OnInit, OnDestroy
     // tslint:disable-next-line:max-line-length
     this.myForm.get('legislations').dirty &&
       (administrativeSanction['legislation'] = this.parseLegislationsFormGroups());
-
+    // swapping legislation with actCode
+    const actTitle = administrativeSanction['legislation'][0]['act'];
```

Comment:
Need to check before indexing here?

Reply:
In all further instances below as well?

Reply:
Would it make sense for this to be a util function? Easier to test and update.

Reply:
created a function in record-utils to take this object and swap values in place. also included error handling. refactored all these add-edit components to use this new function